app-modules/interaction/src/Filament/Resources/InteractionResource/Pages/ListInteractions.php

```diff
@@ -11,34 +11,45 @@
 use Filament\Tables\Columns\TextColumn;
 use Filament\Resources\Pages\ListRecords;
 use Assist\Interaction\Models\Interaction;
+use App\Concerns\FilterTableWithOpenSearch;
 use Filament\Tables\Actions\BulkActionGroup;
 use Filament\Tables\Actions\DeleteBulkAction;
 use Assist\Interaction\Imports\InteractionsImporter;
 use Assist\Interaction\Filament\Resources\InteractionResource;
+use App\Filament\Columns\OpenSearch\TextColumn as OpenSearchTextColumn;
 
 class ListInteractions extends ListRecords
 {
+    use FilterTableWithOpenSearch;
+
     protected static string $resource = InteractionResource::class;
 
     public function table(Table $table): Table
     {
         return $table
             ->columns([
                 IdColumn::make(),
-                TextColumn::make('campaign.name')
-                    ->searchable(),
-                TextColumn::make('driver.name')
-                    ->searchable(),
-                TextColumn::make('division.name')
-                    ->searchable(),
-                TextColumn::make('outcome.name')
-                    ->searchable(),
-                TextColumn::make('relation.name')
-                    ->searchable(),
-                TextColumn::make('status.name')
-                    ->searchable(),
-                TextColumn::make('type.name')
-                    ->searchable(),
+                OpenSearchTextColumn::make('campaign_name')
+                    ->searchable()
+                    ->getStateUsing(fn (Interaction $record) => $record->campaign->name),
```

Comment:
Because this and others like driver, outcome ,division, etc. are probably make more sense as filters I am thinking these names should not be indexed / searchable.

If we don't have filters for these already, then we don't need to add them now. But we should probably only add text columns and searchable for things that don't make more sense as filters like subject and description